### PR TITLE
tikv: fix RowKVClient scan method can't rightly break loop

### LINKED ISSUE #14363: The Scan and ReverseScan method of rawkv client can't respond rightly
## Bug Report

1. What did you do?

I used the rawkv client in my program, when I scan something,  it does not respond to me, even waiting a long time.

The pkg path:

```
github.com/pingcap/tidb/store/tikv
```

Problem functions

```go
func (c *RawKVClient) Scan(startKey, endKey []byte, limit int) {}
func (c *RawKVClient) ReverseScan(startKey, endKey []byte, limit int) {}
```

2. What did you expect to see?

The scan method not respond to me, It was stuck.

3. What did you see instead?

Respond success.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```
github.com/pingcap/tidb/store/tikv@84e4386
```

### PR DESCRIPTION
**Pull request by ldnvnbl**
### What problem does this PR solve? 

fix #14363 

### Check List 

Tests 

 - Unit test
    - The unit test is hard to write for this case, because the scan method will respond when it scanned all data.

 - Integration test
 - Manual test (add detailed scripts or steps below)
   - the tikv cluster has a lot of data
   - scan the key that the cluster not exist
   - I add some log and look the log.

before fix:

image

after fix:
image

 - No code

Release note

 - fix RowKVClient scan method can't rightly break loop

### CODE REVIEW

**Comment by lysu** [comment]
@ldnvnbl  do you  convenience to fire a pr in https://github.com/tikv/client-go/blob/master/rawkv/rawkv.go?

we are plan to move rawkv to tikv/client-go repo and more recommand rawkv user using tikv/client-go

**Comment by ldnvnbl** [comment]
@lysu  OK

File: store/tikv/rawkv.go [hunk 1]
```diff
@@ -313,6 +313,10 @@ func (c *RawKVClient) Scan(startKey, endKey []byte, limit int) (keys [][]byte, v
 		if len(startKey) == 0 {
 			break
 		}
+
+		if len(endKey) > 0 && bytes.Compare(startKey, endKey) > 0 {
```
Comment (disksing) [inline store/tikv/rawkv.go]: I think `bytes.Compare(startKey, endKey) > 0` should be `bytes.Compare(startKey, endKey) >= 0`. And I would like to move the check before sending request:
```
for len(keys) < limit && (len(endKey)==0 || bytes.Compare(startKey, endKey) >= 0) {
  ...
}
```
In this way if startKey >= endKey, we don't need to send any requests.

Comment (ldnvnbl) [reply store/tikv/rawkv.go]: ok

**Comment by ldnvnbl** [comment]
@disksing updated

**Comment by disksing** [comment]
LGTM. Thanks @ldnvnbl 
/cc @lysu

**Comment by ldnvnbl** [comment]
In my local  all tests passed.  so how do I fix the failed Jenkins job ? @qw4990

**Pull request merged**